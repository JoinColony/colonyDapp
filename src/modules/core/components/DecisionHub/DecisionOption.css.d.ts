export const main: string;
export const themeAlt: string;
export const stateDisabled: string;
export const rowContent: string;
export const rowIcon: string;
export const tooltip: string;
