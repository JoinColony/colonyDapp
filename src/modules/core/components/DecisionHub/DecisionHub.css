@import 'variables.css';

.row {
  display: flex;
  padding-right: 16px;
  padding-top: 12px;
  padding-bottom: 12px;
  height: 64px;
  border-bottom: 1px solid var(--grey-blue-0);
}

.rowContent {
  flex: 6;
}

.rowIcon {
  flex: 1;
  padding-left: 16px;
}
