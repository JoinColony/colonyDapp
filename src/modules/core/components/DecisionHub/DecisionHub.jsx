/* @flow */

import React from 'react';
import { injectIntl } from 'react-intl';

import type { MessageDescriptor } from 'react-intl';

import styles from './DecisionHub.css';

import Icon from '../Icon';
import Heading from '../Heading';

const displayName = 'DecisionHub';

type RowProps = {
  title: MessageDescriptor,
  subTitle: MessageDescriptor,
  rowIndex: number,
  icons?: Array<string>,
};

type DecisionProps = {
  rowTitles: Object,
  rowSubTitles: Object,
  icons?: Array<string>,
};

const DecisionOption = ({ title, subTitle, rowIndex, icons }: RowProps) => (
  <div className={styles.row}>
    {icons && (
      <div className={styles.rowIcon}>
        <Icon name={icons[rowIndex]} title={icons[rowIndex]} />
      </div>
    )}
    <div className={styles.rowContent}>
      <Heading
        appearance={{ size: 'small', weight: 'bold', margin: 'small' }}
        text={title}
      />
      <Heading
        appearance={{ size: 'tiny', weight: 'thin', margin: 'small' }}
        text={subTitle}
      />
    </div>
    <Icon name="caret-right" title="caret-right" />
  </div>
);

const DecisionHub = ({ icons, rowTitles, rowSubTitles }: DecisionProps) =>
  Object.keys(rowTitles).map((key, i) => {
    const keys = Object.keys(rowSubTitles);
    const title = rowTitles[key];
    const subTitle = rowSubTitles[keys[i]];

    return (
      <div key={`Link${title.id}`}>
        <DecisionOption
          icons={icons}
          title={title}
          subTitle={subTitle}
          key={`row${title.id}`}
          rowIndex={i}
        />
      </div>
    );
  });

DecisionHub.displayName = displayName;

export default injectIntl(DecisionHub);
