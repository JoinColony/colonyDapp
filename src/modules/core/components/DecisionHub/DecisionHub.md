### Decision Hub Usage
The Decision Hub renders a list of links divided by borders

```js

const rowTitles = {
  superCreativeTitle: {
    id: 'ComponentName.special',
    defaultMessage: 'Rule the world',
  },
  evenMoreCreativeTitle: {
    id: 'ComponentName.exceptional',
    defaultMessage: 'Collaborate with others',
  },
};

const rowSubTitles = {
  superCreativeTitle: {
    id: 'ComponentName.specialSub',
    defaultMessage: 'But be nice',
  },
  evenMoreCreativeTitle: {
    id: 'ComponentName.exceptionalSub',
    defaultMessage: 'And kick ass',
  },
};

<DecisionHub rowTitles={rowTitles} rowSubTitles={rowSubTitles} />
```

Decision Hub with icons for each row

```js

const rowTitles = {
  superCreativeTitle: {
    id: 'ComponentName.special',
    defaultMessage: 'Rule the world',
  },
  evenMoreCreativeTitle: {
    id: 'ComponentName.exceptional',
    defaultMessage: 'Collaborate with others',
  },
};

const rowSubTitles = {
  superCreativeTitle: {
    id: 'ComponentName.specialSub',
    defaultMessage: 'But be nice',
  },
  evenMoreCreativeTitle: {
    id: 'ComponentName.exceptionalSub',
    defaultMessage: 'And kick ass',
  },
};

const icons = ['metamask', 'wallet'];

<DecisionHub rowTitles={rowTitles} rowSubTitles={rowSubTitles} icons={icons}/>
```