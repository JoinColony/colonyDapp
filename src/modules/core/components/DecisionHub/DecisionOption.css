.main {
  display: flex;
  align-items: center;
  padding: 12px 16px 12px 0;
  height: 64px;
  width: 100%;
  border: none;
  border-top: 1px solid var(--grey-blue-0);
  background: none;
  text-align: left;
  cursor: pointer;

  &:only-of-type {
    border-bottom-color: transparent;
  }

  &:last-of-type:not(.themeAlt) {
    border-bottom: 1px solid var(--grey-blue-0);
  }
}

.themeAlt {
  composes: main;
  border: none;
  background-color: white;
}

.rowContent {
  flex: 1;
}

.rowIcon {
  flex: 0 0 50px;
  text-align: center;
}

.tooltip {
  max-width: 270px;
}
