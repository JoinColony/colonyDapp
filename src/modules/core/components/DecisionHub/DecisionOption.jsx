/* @flow */

import React, { Component } from 'react';
import type { MessageDescriptor } from 'react-intl';

import { defineMessages, FormattedMessage } from 'react-intl';

import { getMainClasses } from '~utils/css';

import Icon from '../Icon';
import Link from '../Link';
import { Tooltip } from '../Popover';
import Heading from '../Heading';
import { asField } from '../Fields';

import styles from './DecisionOption.css';

const MSG = defineMessages({
  iconTitle: {
    id: 'DecisionOption.iconTitle',
    defaultMessage: 'Choose this option',
  },
});

type Appearance = {|
  theme?: 'alt',
|};

type Props = {|
  appearance?: Appearance,
  option: {|
    value: string,
    title: MessageDescriptor | string,
    subtitle: MessageDescriptor | string,
    icon?: string,
    tooltip?: MessageDescriptor,
  |},
  link?: string,
  /** @ignore Will be injected by `asField` */
  setValue: (val: string) => void,
|};

const displayName = 'DecisionOption';

class DecisionOption extends Component<Props> {
  makeDecision = () => {
    const {
      option: { value },
      setValue,
    } = this.props;
    setValue(value);
  };

  renderIcon = (tooltip, icon, title) => {
    /* Wrap icon in tooltip wrapper only if tooltip propert exists */
    if (tooltip && icon) {
      return (
        <Tooltip
          placement="left"
          trigger="hover"
          content={
            <span className={styles.tooltip}>
              <FormattedMessage {...tooltip} />
            </span>
          }
        >
          <div className={styles.rowIcon}>
            <Icon name={icon} title={title} appearance={{ size: 'small' }} />
          </div>
        </Tooltip>
      );
    }
    if (icon) {
      return (
        <div className={styles.rowIcon}>
          <Icon name={icon} title={title} />
        </div>
      );
    }
    return null;
  };

  render() {
    const {
      appearance,
      option: { icon, subtitle, title, tooltip },
      link,
    } = this.props;
    const Element = link ? Link : 'button';
    const elmProps = link ? { to: link } : { onClick: this.makeDecision };

    return (
      <Element
        type="submit"
        className={getMainClasses(appearance, styles)}
        {...elmProps}
      >
        {this.renderIcon(tooltip, icon, title)}
        <div className={styles.rowContent}>
          <Heading
            appearance={{ size: 'small', weight: 'bold', margin: 'small' }}
            text={title}
          />
          <Heading
            appearance={{ size: 'tiny', weight: 'thin', margin: 'small' }}
            text={subtitle}
          />
        </div>
        <Icon name="caret-right" title={MSG.iconTitle} />
      </Element>
    );
  }
}

DecisionOption.displayName = displayName;

export default asField({ initialValue: '' })(DecisionOption);
