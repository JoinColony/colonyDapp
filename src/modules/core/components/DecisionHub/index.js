/* @flow */

export { default } from './DecisionHub.jsx';
