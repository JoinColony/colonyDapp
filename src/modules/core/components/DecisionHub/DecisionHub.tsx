import React from 'react';

import DecisionOption, { DecisionOptionType } from './DecisionOption';

const displayName = 'DecisionHub';

interface Props {
  options: DecisionOptionType[];
  name: string;
}

const DecisionHub = ({ options, name }: Props) => (
  <div data-test="hubOptions">
    {options.map(option => (
      <DecisionOption name={name} option={option} key={`row-${option.value}`} />
    ))}
  </div>
);

DecisionHub.displayName = displayName;

export default DecisionHub;
