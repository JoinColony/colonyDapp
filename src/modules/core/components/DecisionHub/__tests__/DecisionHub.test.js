/* eslint-env jest */

import React from 'react';
import toJson from 'enzyme-to-json';

import { shallowWithIntl } from 'testutils';

import DecisionHub from '../DecisionHub.jsx';

describe('DecisionHub component', () => {
  test('Renders initial component', () => {
    const rowTitles = {
      superCreativeTitle: {
        id: 'ComponentName.special',
        defaultMessage: 'Rule the world',
      },
      evenMoreCreativeTitle: {
        id: 'ComponentName.exceptional',
        defaultMessage: 'Collaborate with others',
      },
    };

    const rowSubTitles = {
      superCreativeTitle: {
        id: 'ComponentName.specialSub',
        defaultMessage: 'But be nice',
      },
      evenMoreCreativeTitle: {
        id: 'ComponentName.exceptionalSub',
        defaultMessage: 'And kick ass',
      },
    };
    const wrapper = shallowWithIntl(
      <DecisionHub rowTitles={rowTitles} rowSubTitles={rowSubTitles} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
