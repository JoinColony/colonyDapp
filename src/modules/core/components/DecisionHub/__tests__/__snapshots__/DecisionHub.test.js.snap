// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecisionHub component Renders initial component 1`] = `
<DecisionHub
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  rowSubTitles={
    Object {
      "evenMoreCreativeTitle": Object {
        "defaultMessage": "And kick ass",
        "id": "ComponentName.exceptionalSub",
      },
      "superCreativeTitle": Object {
        "defaultMessage": "But be nice",
        "id": "ComponentName.specialSub",
      },
    }
  }
  rowTitles={
    Object {
      "evenMoreCreativeTitle": Object {
        "defaultMessage": "Collaborate with others",
        "id": "ComponentName.exceptional",
      },
      "superCreativeTitle": Object {
        "defaultMessage": "Rule the world",
        "id": "ComponentName.special",
      },
    }
  }
/>
`;
